src/services/observables/tokens.ts

```diff
@@ -328,6 +328,9 @@ export const pools$ = combineLatest([
         const reserveTokenTwo = tokens.find(
           (t) => t.address === pool.reserves[1].address
         );
+
+        if (!reserveTokenOne || !reserveTokenTwo) return null;
```

Comment:
1. You can check whether `reserveTokenOne` is found before searching for `reserveTokenTwo` and save unnecessary check.
2. Also, this appears to be a very expensive concatenated loop - for each pool we are iterating over a token list twice and perform string comparison. We can create a data structure - i.e. a map from reserve address to token. It will take a single iteration to build - and then get in O(1) - compared to all these iterations over the token list.
3. Lastly, after this check you know both reserve tokens are defined - so you can avoid checks like in line 343